src/config.rs

```diff
@@ -0,0 +1,42 @@
+use crate::utils;
+
+use dirs::home_dir;
+
+pub struct Config {
+    data: toml::value::Table,
+}
+
+impl Config {
+    /// Initialize the Config struct
+    pub fn initialize() -> Config {
+        if let Some(file_data) = Config::config_from_file() {
+            return Config { data: file_data };
+        }
+
+        Config {
+            data: toml::value::Table::new(),
+        }
+    }
+
+    /// Create a config from a starship configuration file
+    fn config_from_file() -> Option<toml::value::Table> {
+        let file_path = home_dir()?.join(".config/starship.toml");
+        let toml_content = utils::read_file(&file_path.to_str()?).ok()?;
```

Comment:
Good call! 👍 
Not something we have to address in this PR, but I think we may want to think about how we want to elegantly show warnings and alerts (configuration error, update available, etc). Because every prompt instance is stateless, it would print the error _every time_. 🤔 

What would be the ideal way to inform the user that something is wrong with their configuration without getting in their way as they address it?
IMO, the ideal solution would be to print a warning at the start of each session (3 hrs since the prompt was last used would be a fresh instance, for instance).